# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1521684851 18000
#      Wed Mar 21 21:14:11 2018 -0500
# Branch stable
# Node ID 7f1d1a1ca22ef91bf18654b4b9dbb17bd3ab21d8
# Parent  57d99b5e643a1487e74a2fbccd43469c91edbff1
Remove NoFocus policy from Variable Editor and restore highlight (bug #53410)

* variable_editor.cc (variable_editor::variable_editor): Remove all lines
  regarding setFocusPolicy() thereby using the octave_dock_widget default.

diff --git a/libgui/src/variable-editor.cc b/libgui/src/variable-editor.cc
--- a/libgui/src/variable-editor.cc
+++ b/libgui/src/variable-editor.cc
@@ -990,25 +990,22 @@ namespace octave
       m_table_colors (),
       m_current_focus_vname (""),
       m_hovered_focus_vname ("")
   {
     setObjectName ("VariableEditor");
     set_title (tr ("Variable Editor"));
     setStatusTip (tr ("Edit variables."));
     setWindowIcon (QIcon (":/actions/icons/logo.png"));
-    setFocusPolicy (Qt::NoFocus);
     setAttribute (Qt::WA_AlwaysShowToolTips);
 
     // Tool Bar.
 
     construct_tool_bar ();
     m_main->addToolBar (m_tool_bar);
-    m_main->setFocusPolicy (Qt::NoFocus);
-    setFocusPolicy (Qt::NoFocus);
 
     // Colors.
 
     for (int i = 0; i < resource_manager::varedit_color_chars ().length (); i++)
       m_table_colors.append (QColor (Qt::white));
 
     // Use an MDI area that is shrunk to nothing as the central widget.
     // Future feature might be to switch to MDI mode in which the dock
